src/legends/color.js

```diff
@@ -0,0 +1,158 @@
+import {swatches} from "./swatches.js";
+import {scale} from "../scales.js";
+import {create, scaleLinear, quantize, interpolate, interpolateRound, quantile, range, format, scaleBand, axisBottom} from "d3";
+
+export function colorLegend(descriptor, options) {
+  switch(descriptor.type) {
+    case "ordinal":
+    case "categorical":
+      return swatches(scale(descriptor), options);
+  }
+  return legend(scale(descriptor), {
+    title: descriptor.label,...options
+  });
+}
+
+function legend(color, {
+  title,
+  tickSize = 6,
+  width = 320, 
```

Comment:
```suggestion
  width = 240, 
```

320 feels a little wide as a default. Should probably also be capped at the width of the chart in the legend: true case.